spring-boot-sample-project/src/main/java/com/example/demo/controller/UserController.java

```diff
@@ -23,9 +23,9 @@
         return ResponseEntity.ok(userList);
     }
 
-    @GetMapping("/user/{name}")
-    public ResponseEntity<User> getUser(@PathVariable(value="name") String userName){
-        User user = userService.getAUser(userName);
+    @GetMapping("/user/{id}")
+    public ResponseEntity<Optional<User>> getUser(@PathVariable(value="id") int id){
+        Optional<User> user = userService.getAUser(id);
```

Comment:
optional should never returned as response. It should be resolved before that
your code should:
```java
if (user.isEmpty()) {
  return new ResponseEntity<User>(HttpStatus.NOT_FOUND)
}

return ResponseEntity.ok(user.get())
```